frontend/src/components/widgets/DataFrame/hooks/useDataLoader.test.ts

```diff
@@ -0,0 +1,378 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from "react"
+
+import { renderHook } from "@testing-library/react-hooks"
+import { GridCellKind } from "@glideapps/glide-data-grid"
+
+import { Quiver } from "src/lib/Quiver"
+import { Arrow as ArrowProto } from "src/autogen/proto"
+import { UNICODE } from "src/lib/mocks/arrow"
+import {
+  BaseColumn,
+  ObjectColumn,
+  TextColumn,
+  BooleanColumn,
+  CategoricalColumn,
+  ListColumn,
+  NumberColumn,
+  RangeColumn,
+  ImageColumn,
+  ChartColumn,
+  UrlColumn,
+  ColumnCreator,
+  isErrorCell,
+} from "src/components/widgets/DataFrame/columns"
+import EditingState from "src/components/widgets/DataFrame/EditingState"
+
+import useDataLoader, {
+  ColumnConfigProps,
+  applyColumnConfig,
+  getColumnConfig,
+  getColumnType,
+  INDEX_IDENTIFIER,
+  COLUMN_POSITION_PREFIX,
+} from "./useDataLoader"
+
+const MOCK_COLUMNS: BaseColumn[] = [
+  NumberColumn({
+    id: "index_col",
+    title: "",
+    indexNumber: 0,
+    arrowType: {
+      pandas_type: "int64",
+      numpy_type: "int64",
+    },
+    isEditable: false,
+    isHidden: false,
+    isIndex: true,
+    isStretched: false,
+  }),
+  NumberColumn({
+    id: "column_1",
+    title: "column_1",
+    indexNumber: 1,
+    arrowType: {
+      pandas_type: "int64",
+      numpy_type: "int64",
+    },
+    isEditable: false,
+    isHidden: false,
+    isIndex: false,
+    isStretched: false,
+  }),
+  TextColumn({
+    id: "column_2",
+    title: "column_2",
+    indexNumber: 2,
+    arrowType: {
+      pandas_type: "unicode",
+      numpy_type: "object",
+    },
+    isEditable: false,
+    isHidden: false,
+    isIndex: false,
+    isStretched: false,
+  }),
+]
+
+describe("applyColumnConfig", () => {
```

Comment:
Can we write a TODO / testcase that passes in an undefined columnConfig and just check columnConfig === undefined?